### PR TITLE
add dream dataset

### PR DESCRIPTION
**Pull request by patil-suraj**
Adding Dream: a Dataset and for Dialogue-Based Reading Comprehension

More details:
https://dataset.org/dream/
https://github.com/nlpdata/dream

### CODE REVIEW

**Review by lhoestq — COMMENTED** [review]
Awesome good job !

Could you also add a dataset card using the template guide here : https://github.com/huggingface/datasets/blob/master/templates/README_guide.md
If you can't fill some fields then just leave `[N/A]`

**Comment by yjernite** [comment]
> Awesome good job !
> 
> Could you also add a dataset card using the template guide here : https://github.com/huggingface/datasets/blob/master/templates/README_guide.md
> If you can't fill some fields then just leave `[N/A]`

Quick amendment: `[N/A]` is for fields that are not relevant: if you can't find the information just leave `[More Information Needed]`

**Comment by patil-suraj** [comment]
@lhoestq since datset cards are optional for this sprint I'll add those later. Good for merge.

**Comment by lhoestq** [comment]
Indeed we only require the tags to be added now (the yaml part at the top of the dataset card).
Could you add them please ?
You can find more infos here : https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#tag-the-dataset-and-write-the-dataset-card

**Comment by patil-suraj** [comment]
@lhoestq added tags, I'll fill rest of the info after current sprint :)

**Comment by patil-suraj** [comment]
The tests are failing tests for other datasets, not this one.

**Comment by patil-suraj** [comment]
@lhoestq  could you tell me why these tests are failing, they don't seem related to this PR.

File: datasets/dream/dream.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                {
+                    "id": datasets.Value("string"),
+                    "dialogue": datasets.features.Sequence(datasets.Value("string")),
+                    "questions": [
+                        {
+                            "question": datasets.Value("string"),
+                            "choice": datasets.features.Sequence(datasets.Value("string")),
+                            "answer": datasets.Value("string"),
+                        },
+                    ],
```
Comment (lhoestq) [inline datasets/dream/dream.py:89]: We can make this more practical to use by flattening the nested structure (like other datasets like coqa or com_qa)
```
question: datasets.Value("string"),
choices: datasets.Sequence(datasets.Value("string")),
answer: datasets.Value("string"),
```
this way we have one question per example in the dataset, which is convenient to train models.

What do you think ?

**Review by lhoestq — COMMENTED** [review]
Clean !

I have one suggestion about the feature types. I think we can make them more practical and consistent with the other datasets.

File: datasets/dream/dream.py (hunk 1 above)
Comment (patil-suraj) [reply datasets/dream/dream.py:89]: Makes sense !

**Review by lhoestq — APPROVED** [review]
Thanks !

**Pull request merged**